pingcap/tidb-operator
# lightning: support lightning use IAM

### What problem does this PR solve? 
add serviceAccount for config
### What is changed and how does it work?
add service Account config to pod template
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1. install an importer in tidb cluster

2. add annotation in annotation field
```
annotations: {"iam.amazonaws.com/role": "arn:aws:iam::123456789012:role/user"}
```
and set a seviceAccount in values.yaml

3. follow [doc](https://pingcap.com/docs-cn/stable/reference/tools/tidb-lightning/deployment/) to deploy lightning in tidb cluster and use remote S3 backups to restore , ignore accesskey and 
 secret key in field `dataSource.remote.secretName`

4. check wether initContainer can download file from S3 and the serviceAccount is attach to the tidb-lightning pod

Code changes

 - Has Go code change
